auth/src/providers/AzureSubscriptionProviderBase.ts

```diff
@@ -0,0 +1,417 @@
+        }
+
+        return this.refreshSuggestedEmitter.event(callback, thisArg, disposables);
+    }
+
+    protected fireRefreshSuggestedIfNeeded(reason: RefreshSuggestedEvent): void {
+        if (this.suppressRefreshSuggestedEvents || Date.now() < this.lastRefreshSuggestedTime + EventDebounce) {
+            // Suppress and/or debounce events to avoid flooding
+            return;
+        }
+
+        this.log('Firing onRefreshSuggested event');
+        this.lastRefreshSuggestedTime = Date.now();
+        this.refreshSuggestedEmitter.fire(reason);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async signIn(tenant?: Partial<TenantIdAndAccount>, options: SignInOptions = DefaultSignInOptions): Promise<boolean> {
+        const prompt = options.promptIfNeeded ?? DefaultSignInOptions.promptIfNeeded;
+
+        if (prompt) {
+            // If interactive, suppress without timeout until sign in is done (it can take a while when done interactively)
+            this.suppressRefreshSuggestedEvents = true;
+        } else {
+            // If silent, suppress with normal timeout
+            this.silenceRefreshEvents();
+        }
+
+        const session = await getSessionFromVSCode(
+            undefined,
+            tenant?.tenantId,
+            {
+                account: tenant?.account,
+                clearSessionPreference: options.clearSessionPreference ?? DefaultSignInOptions.clearSessionPreference,
+                createIfNone: prompt,
+                silent: !prompt,
+            }
+        );
+
+        if (prompt) {
+            // Interactive sign in can take a while, so silence events for a bit longer
+            this.silenceRefreshEvents();
+        }
+
+        return !!session;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async getAvailableSubscriptions(options: GetAvailableSubscriptionsOptions = DefaultOptions): Promise<AzureSubscription[]> {
+        try {
+            const availableSubscriptions: AzureSubscription[] = [];
+
+            const tenantListLimiter = new Limiter<void>(TenantListConcurrency);
+            const tenantListPromises: Promise<void>[] = [];
+
+            const subscriptionListLimiter = new Limiter<void>(SubscriptionListConcurrency);
+            const subscriptionListPromisesFlat: Promise<void>[] = [];
+
+            let tenantsProcessed = 0;
+            const maximumTenants = options.maximumTenants ?? DefaultOptions.maximumTenants;
+
+            const accounts = await this.getAccounts(options);
+
+            for (const account of accounts) {
+                this.throwIfCancelled(options.token);
+                tenantListPromises.push(tenantListLimiter.queue(async () => {
+                    try {
+                        if (tenantsProcessed >= maximumTenants) {
+                            this.logForAccount(account, `Skipping account because maximum tenants of ${maximumTenants} has been reached`);
+                            return;
+                        }
```

Comment:
As implemented, there is no guarantee that these 10 tenants are included in the tenant filters. As a result, if (for example) the first 10 tenants are not in the filter, then no subscriptions will ever be returned. Need to fix this.

Reply:
Maybe? Actually I am having a hard time determining by reading! Tells me the code needs refactoring to be clearer.